climate-data-ca/resources/js/analyze-functions.js

```diff
@@ -1885,6 +1887,318 @@
             }
         }
 
+
+        function buildShareableURL(form_inputs_dict) {
+            // Cleaning values
+            form_inputs_dict['submit_url_var'] = submit_url_var
+            // Passing by encodingURI makes it easier to read on the reception
+            form_inputs_dict['form_thresholds'] = encodeURI(JSON.stringify(form_thresholds))
+
+            // Compressing the [lat,long] points to a compressed string
+            let latAray = form_inputs['lat'].split(',')
+            let lonArray = form_inputs['lon'].split(',')
+            form_inputs_dict["compressedPoints"] = getEncodedPoints(latAray,lonArray)
+
+            // Emptying the lat & lon arrays because of the use of the compressed string to shorten the shareable url
+            form_inputs_dict['lat'] = ""
+            form_inputs_dict['lon'] = ""
+
+            // Building the shareable url
+            // Convert dict to url &param=value like
+            let params = new URLSearchParams(form_inputs_dict)
+
+            let baseUrl = document.URL.split('?')[0]
+
+            let shareableURL = baseUrl + "?" + params
+
+            updateShareableURL(shareableURL)
+        }
+
+        function updateShareableURL(newUrl) {
+            // Update browser url
+            window.history.pushState('updateAnalyzeFormUrl', 'new_url', newUrl)
+
+            // Update copy link button
+            $("a[id='shareableURL']").attr('data-share-url', newUrl)
+        }
+
+        
+        function readURL() {
+            console.log(data_url)
+            let params = window.location.search
+            params = params.split('?').pop() // Remove ? at beginning
+            params = params.split('&') // Split into a key=value list
+
+            // Parse the params list into a {key:value} dict
+            let paramsDict = {}
+            for (var p of params) {
+                let keyValue = p.split('=')
+                let key = keyValue[0]
+                let value = decodeURIComponent(keyValue[1])
+                paramsDict[key] = value
+            }
+
+            // Parse tresholds value that have been encoded
+            paramsDict["form_thresholds"] = JSON.parse(decodeURI(paramsDict["form_thresholds"]))
```

Comment:
I'm getting an exception here when switching language. Google Analytics appends something to the query string. ex.:https://dev-fr.climatedata.ca/analyser/?_gl=1*19kjqll*_ga*MTIyOTg2MTI4Ny4xNTYyNTg3MjE4*_ga_CPGH39EQSH*MTY4NjA1OTk4Mi4yMTcuMS4xNjg2MDYwMDA5LjAuMC4w